test/01_cond_field/process.py

```diff
@@ -1,194 +1,111 @@
-        ])
-
-        pbs_config = dict(
-            n_cores=1,
-            n_nodes=1,
-            select_flags=['cgroups=cpuacct'],
-            mem='128mb',
-            queue='charon_2h',
-            home_dir='/storage/liberec3-tul/home/martin_spetlik/',
-            pbs_process_file_dir='/auto/liberec3-tul/home/martin_spetlik/MLMC_new_design/src/mlmc')
-
-        sampling_pool = SamplingPoolPBS(job_weight=200000, job_count=0, work_dir=self.work_dir)
-
-        # Set pbs config, flow123d, gmsh, ...
+        # Set pbs config, flow123d, gmsh, ..., random fields are set in simulation class
         self.set_environment_variables()
-        output_dir = os.path.join(self.work_dir, "output_{}".format(n_levels))
-        # remove existing files
-        if clean:
-            self.rm_files(output_dir)
 
-        # Init pbs object
-        self.create_pbs_object(output_dir, clean)
+        # Create Pbs sampling pool
+        sampling_pool = self.create_pbs_sampling_pool()
+
+        #sampling_pool = OneProcessPool(work_dir=self.work_dir)  # Everything runs in one process
+        #sampling_pool = ProcessPool(n_processes=4, work_dir=self.work_dir)  # Simulations run in different processes
 
         simulation_config = {
-            'env': dict(flow123d=self.flow123d, gmsh=self.gmsh, pbs=self.pbs_obj),  # The Environment.
-            'output_dir': output_dir,
-            'fields': fields,  # correlated_field.FieldSet object
-            'yaml_file': os.path.join(self.work_dir, '01_conductivity.yaml'),  # The template with a mesh and field placeholders
-            'sim_param_range': self.step_range,  # Range of MLMC simulation parametr. Here the mesh step.
-            'geo_file': os.path.join(self.work_dir, 'square_1x1.geo'),  # The file with simulation geometry (independent of the step)
+            'work_dir': self.work_dir,
+            'env': dict(flow123d=self.flow123d, gmsh=self.gmsh, gmsh_version=1),  # The Environment.
+            'yaml_file': os.path.join(self.work_dir, '01_conductivity.yaml'),
+            # The template with a mesh and field placeholders
+            'sim_param_range': step_range,  # Range of MLMC simulation parametr. Here the mesh step.
+            'geo_file': os.path.join(self.work_dir, 'square_1x1.geo'),
+            # The file with simulation geometry (independent of the step)
             # 'field_template': "!FieldElementwise {mesh_data_file: \"${INPUT}/%s\", field_name: %s}"
             'field_template': "!FieldElementwise {mesh_data_file: \"$INPUT_DIR$/%s\", field_name: %s}"
         }
 
-        FlowProcSim.total_sim_id = 0
-
-        self.options['output_dir'] = output_dir
-        mlmc_obj = mlmc.mlmc.MLMC(n_levels, FlowProcSim.factory(self.step_range, config=simulation_config, clean=clean),
-                                  self.step_range, self.options)
+        # Create simulation factory
+        simulation_factory = FlowSim(step_range, config=simulation_config, clean=clean)
```

Comment:
Do we need to pass step_range to FlowSim, as a factory it should not be the case.